spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -183,6 +193,7 @@ public Statistics estimateStatistics() {
     return estimateStatistics(SnapshotUtil.latestSnapshot(table, branch));
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
Let's try to reduce the complexity rather than suppressing the warning

General this means

Removing nesting
Extracting sub-functions out
Avoiding early exits